routes/api/listVersions.js

```diff
@@ -1,6 +1,190 @@
 const express = require("express");
 const router = express.Router();
 
-router.get("/test", (req, res) => res.json({ msg: "This is the lists versions route" }));
+const ListVersion = require("../../models/ListVersion");
+const TextString = require("../../models/TextString");
+const Review = require("../../models/Review");
+
+// Show
+router.get("/:id", (req, res) => {
+  ListVersion.findById(req.params.id)
+    .populate("text")
+    .populate("reviews")
+    .populate({
+      path: "listId",
+      populate: {
+        path: "name",
+        path: "description",
+      },
+    })
+    .then((ListVersion) => {
+      if (ListVersion) {
+        res.json(ListVersion);
+      } else {
+        res.status(404).json("No Such Version");
+      }
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Index
+router.get("/", (req, res) => {
+  ListVersion.find({})
+    .populate("text")
+    .populate("reviews")
+    .populate({
+      path: "listId",
+      populate: {
+        path: "name",
+        path: "description",
+      },
+    })
+    .then((ListVersions) => {
+      res.json(ListVersions);
+    })
+    .catch((err) => {
+      console.log(err)
+      res.status(404).json(err)});
+});
+
+// All Versions of List
+router.get("/:listId", (req, res) => {
+  ListVersion.find({ listId: req.params.listId})
+    .populate("text")
+    .populate("reviews")
+    .populate({
+      path: "listId",
+      populate: {
+        path: "name",
+        path: "description",
+      },
+    })
+    .then((ListVersions) => {
+      res.json(ListVersions);
+    })
+    .catch((err) => {
+      console.log(err);
+      res.status(404).json(err);
+    });
+});
+
+// Create
+router.post("/", async (req, res) => {
+  let { listId, text, month, year, reviews } = req.body;
+
+  text = await TextString.create(text);
+
+  let listVersion = new ListVersion({
+    listId,
+    text,
+    month,
+    year,
+  });
+
+  ListVersion.create(listVersion)
+    .then((listVersion) => res.json(listVersion))
+    .catch((err) => res.json(err));
+});
+
+// Create New Version
+router.post("/:id", async (req, res) => {
+
+  await ListVersion.findById(req.params.id)
+    .populate("text")
+    .populate({
+      path: "reviews",
+      populate: {
+        path: "description",
+      },
+    })
+    .exec(async function (err, listVersion) {
+      if (err) console.log(err);
+
+      listVersion._id = undefined;
+      listVersion.isNew = true;
+
+      let oldText = await TextString.findById(listVersion.text._id)
+
+      listVersion.text = await TextString.create({
+        hebrew: oldText.hebrew,
+        english: oldText.english
+      })
+
+      let newReviews = [];
+      for (let i = 0; i < listVersion.reviews.length; i++) {
```

Comment:
Rule of thumb - if you're using a `for` loop in JS, you're probably doing something wrong.
This is a good example of when you should use [lodash](https://www.npmjs.com/package/lodash).
It has awesome [documentation](https://lodash.com/docs/4.17.15). 
After you add it to your packages, you import it like this:
`const _ = require('lodash');`

For this logic, you want to use `_.map`